internal/pkg/runtime/engines/singularity/cleanup.go

```diff
@@ -25,6 +25,13 @@ import (
 func (engine *EngineOperations) CleanupContainer(fatal error, status syscall.WaitStatus) error {
 	sylog.Debugf("Cleanup container")
 
+	if engine.EngineConfig.GetDeleteImage() {
+		image := engine.EngineConfig.GetImage()
+		if err := os.RemoveAll(image); err != nil {
+			sylog.Errorf("failed to delete container image %s: %s", image, err)
```

Comment:
Are we intentionally calling `Errorf` and not exiting here? This makes it possible to fail the image deletion and still return a 0 exit code

Reply:
Yes it's best effort, we can't return error here and interrupt all other cleanup steps like cgroups, network and let garbage on node